pkg/util/equality/podset.go

```diff
@@ -26,8 +26,8 @@ import (
 
 // TODO: Revisit this, maybe we should extend the check to everything that could potentially impact
 // the workload scheduling (priority, nodeSelectors(when suspended), tolerations and maybe more)
-func comparePodTemplate(a, b *corev1.PodSpec) bool {
-	if !equality.Semantic.DeepEqual(a.Tolerations, b.Tolerations) {
+func comparePodTemplate(a, b *corev1.PodSpec, ignoreTolerations bool) bool {
+	if !ignoreTolerations && !equality.Semantic.DeepEqual(a.Tolerations, b.Tolerations) {
```

Comment:
Currently this PR only ignores the equality check on tolerations, while the [comment](https://github.com/kubernetes-sigs/kueue/issues/2213) indicates we should also relax the validation for nodeSelectors. How confident we are tolerations are enough? 
IIUC the `autoscaling.gke.io/provisioning-request` nodeSelector can be added as well based on the annotation.

Reply:
The node selectors are not part of the equivalency checks.

Reply:
Got it, thanks for clarifying. Do we have this covered in an integration test?

Reply:
Anyway we can probably defer this to a follow up manual testing